test/models/document_test.rb

```diff
@@ -1,7 +1,25 @@
 require "test_helper"
 
 class DocumentTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test "does not accept files larger than 50 MiB" do
+    document = Document.new(title: "Test document")
+    file = mock_file(size: 50.megabytes, content_type: "application/pdf", filename: "testfile.pdf")
+    document.file.attach(io: file)
+
+    assert_not document.valid?
+  end
 end
```

Comment:
A unit test here might not give us a lot of value. I would recommend looking at creating an integration test on the `ContractsController` instead, where we upload some example PDF file stored in `test/fixtures/files`, then we can just use a `fixture_file_upload("my.pdf", "application/pdf")` to do the test of this and other Model validators.

- https://guides.rubyonrails.org/v7.0/active_storage_overview.html#testing
- https://api.rubyonrails.org/v7.0.8.6/classes/ActionDispatch/TestProcess/FixtureFile.html

System Test could also work, but its going to run much faster as an integration test.

Reply:
That's what I said all along...

Reply:
Ah I am sorry for not picking up on that sooner! Please feel free to refactor this into what you think is best.